### ISSUE TITLE
How to translate a Map?

### ISSUE THREAD

**Issue by micobarac** [ngneat/transloco#272]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[x] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

I am trying to translate multiple keys with each param referencing the linked key, but I cannot find a way how to do it.

I am interested in using **selectTranslate** method in particular.

## Expected behavior

Actually, I would like to be able to link every key in array of keys with a separate HashMap of params.

## Minimal reproduction of the problem with instructions

I have `columns`, with each column supporting different `operation`, like `min, max, sum, avg`.

Columns:

```typescript
columns: Column[] = [
  {
    name: 'queue',
    operation: Operation.SUM
  },
  {
    name: 'wp',
    operation: Operation.AVG
  },
  {
    name: 'wr',
    operation: Operation.SUM
  },
  {
    name: 'tps',
    operation: Operation.SUM
  },
  {
    name: 'peak',
    operation: Operation.MAX
  }
];

const phrases: Map<string, HashMap> = new Map(columns.map((column) => [`Performance.Columns.${column.name}`, { operation: column.operation }]));

const translation = this.translocoService.selectTranslate(phrases);
```

Translation file:

```json
  "Performance": {
    "Title": "Performance",
    "ChartTitle": "Summed up values",
    "Period": "Period",
    "Protocol": "Protocol",
    "Columns": {
      "queue": "Queue ({{ operation }})",
      "tps": "Transactions per segment ({{ operation }})",
      "wp": "Workers processing ({{ operation }})",
      "wr": "Workers running ({{ operation }})",
      "peak": "Peak ({{ operation }})"
    }
  }
```

## What is the motivation / use case for changing the behavior?

Have a possibility to pass a different param for each key.

## Environment

<pre><code>
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/

Angular CLI: 9.1.1
Node: 13.10.1
OS: darwin x64

Angular: 9.1.1
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
Ivy Workspace: Yes

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.1
@angular-devkit/build-angular     0.901.1
@angular-devkit/build-optimizer   0.901.1
@angular-devkit/build-webpack     0.901.1
@angular-devkit/core              9.1.1
@angular-devkit/schematics        9.1.1
@angular/cdk                      9.2.0
@angular/flex-layout              9.0.0-beta.29
@angular/material                 9.2.0
@ngtools/webpack                  9.1.1
@schematics/angular               9.1.1
@schematics/update                0.901.1
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: v13.10.1
- Platform:  Mac, Linux, Windows
</code></pre>

**Comment by shaharkazaz** [comment]
@micobarac Seems like you are looking for [selectTranslateObject](https://ngneat.github.io/transloco/docs/translation-api/#selecttranslationobject) method, let me know if that's what you meant.

**Comment by micobarac** [comment]
@shaharkazaz , no, both `selectTranslate` and `selectTranslateObject` have the following definitions:

```typescript
selectTranslate<T = any>(key: TranslateParams, params?: HashMap, lang?: string | TranslocoScope, _isObject?: boolean): Observable<T>;

selectTranslateObject<T = any>(key: TranslateParams, params?: HashMap, lang?: string): Observable<T>;
```

Keys are separated from params. I am talking about a single object, containing a Map of keys with corresponding params. Something like this:

```typescript
selectTranslate<T = any>(keyParams: Map, lang?: string | TranslocoScope, _isObject?: boolean): Observable<T>;

selectTranslateObject<T = any>(keyParams: Map, lang?: string): Observable<T>;
```

**Comment by shaharkazaz** [comment]
@micobarac released in v2.17.0 🙂
